src/call/pairwise.rs

```diff
@@ -189,7 +189,15 @@ pub fn call<A, B, P, M, R, W, X, F>(
             for (i, event) in events.iter().enumerate() {
                 for (j, pileup) in pileups.iter().enumerate() {
                     let p = if let &Some(ref pileup) = pileup {
-                        Some(pileup.posterior_prob(&event.af_case, &event.af_control))
+                        // if either of the pileups is empty, no valid probability could be
+                        // calculated for it -- and in SingleCellBulkModel's joint_prob() and map()
+                        // this would result in a division by zero and thus a program crash
+                        if  !pileup.case_observations().is_empty() &&
+                            !pileup.control_observations().is_empty() {
+                                Some(pileup.posterior_prob(&event.af_case, &event.af_control))
+                            } else {
+                                None
+                            }
```

Comment:
You are absolutely right.

So the clean solution is to include proper priors (one each for the bulk and the single cell sample, as these are distinct in the SingleCellBulkModel) and then handle the case that either `n_obs_s` or `n_obs_bulk` is zero separately. For the single cell sample, this is straightforward, as we are always testing across the cases 0, 0.5 and 1, and we then just return the priors for those. For the bulk sample this is more complicated, as we discretise to the possible allele frequencies, given the total coverage `n_obs_bulk`. But then this approach doesn't make sense for low total coverages, as the Event space with its ranges will be more finely resolved than the allele frequencies we look at. So maybe we should go back to a fixed granularity, as you have it for the TumorNormalModel, or at least have a minimum (and maybe maximum) resolution, e.g. at least 11 different allele frequencies from 0.0..1.0. Any thoughts? Any numerical insights?